test/webdriverio/test/test_setup.ts

```diff
@@ -418,6 +420,75 @@ export async function tabNavigateForward(browser: WebdriverIO.Browser) {
   await browser.pause(PAUSE_TIME);
 }
 
+/**
+ * Navigates backward to the test page's previous tab stop.
+ *
+ * @param browser The active WebdriverIO Browser object.
+ */
+export async function tabNavigateBackward(browser: WebdriverIO.Browser) {
+  await browser.keys([webdriverio.Key.Shift, webdriverio.Key.Tab]);
+  await browser.pause(PAUSE_TIME);
+}
+
+/**
+ * Sends the keyboard event for arrow key left.
+ *
+ * @param browser The active WebdriverIO Browser object.
+ * @param times The number of times to repeat the key press (default is 1).
+ */
+export async function keyLeft(browser: WebdriverIO.Browser, times = 1) {
+  await sendKeyAndWait(browser, webdriverio.Key.ArrowLeft, times);
+}
+
+/**
+ * Sends the keyboard event for arrow key right.
+ *
+ * @param browser The active WebdriverIO Browser object.
+ * @param times The number of times to repeat the key press (default is 1).
+ */
+export async function keyRight(browser: WebdriverIO.Browser, times = 1) {
+  await sendKeyAndWait(browser, webdriverio.Key.ArrowRight, times);
+}
+
+/**
+ * Sends the keyboard event for arrow key up.
+ *
+ * @param browser The active WebdriverIO Browser object.
+ * @param times The number of times to repeat the key press (default is 1).
+ */
+export async function keyUp(browser: WebdriverIO.Browser, times = 1) {
+  await sendKeyAndWait(browser, webdriverio.Key.ArrowUp, times);
+}
+
+/**
+ * Sends the keyboard event for arrow key down.
+ *
+ * @param browser The active WebdriverIO Browser object.
+ * @param times The number of times to repeat the key press (default is 1).
+ */
+export async function keyDown(browser: WebdriverIO.Browser, times = 1) {
+  await sendKeyAndWait(browser, webdriverio.Key.ArrowDown, times);
+}
+
+/**
+ * Sends the specified key for the specified number of times, waiting between
+ * each key press to allow changes to keep up.
+ *
+ * @param browser The active WebdriverIO Browser object.
+ * @param key The WebdriverIO representative key value to press.
+ * @param times The number of times to repeat the key press.
+ */
+async function sendKeyAndWait(
+  browser: WebdriverIO.Browser,
+  key: string,
```

Comment:
Any reason not to make this accept everything that `browser.keys` does?  Then you could call it from `tabNavigateForward` and `tabNavigateBackward` (and other places too, if exported).

Reply:
I'm not opposed, but I think I'll do that in a follow-up if there's no concerns there (since I'd rather move everything over and that's a bit of a scope expansion for this PR).

I also don't think I want to exactly mimic the contract of `browser.keys` since it can do some inadvertent key repeats (which we actually use elsewhere), but allowing it to include meta keys is a good point.